simple

```diff
@@ -220,16 +232,46 @@ docker)
 
 ###############################################################################
 validate)
+    (
+	[ -e $cli_conf_dir ] || (
+            mkdir $cli_conf_dir
+        ) || exit
+
+        [ -e $installed_pip ] || (
+            yum install -y python-pip && pip install --upgrade pip &&
+            touch $installed_pip
+        ) || exit
+
+        [ -e $installed_virtualenv ] || (
+            yum install -y python-virtualenv && touch $installed_virtualenv
+        ) || exit
+
+	[ -e $installed_git ] || (
+	    yum install -y git && touch $installed_git
+	) || exit
+	echo $validator_dir
+        [ -e $installed_validator ] || (
+            mkdir -p $validator_dir &&  cd $validator_dir && pwd &&
+	    git clone "$validator_url" $validator_dir &&
+	    virtualenv $validator_env && source $validator_env/bin/activate &&
+	    pip install -r requirements.txt && pip install $validator_dir &&
+	    deactivate && touch $installed_validator
```

Comment:
is the deactivate still needed?

Reply:
Yup, the validation engine is installed in the currently active $validator_env (line 256) which needs to be deactivated after the installation.
For the YAML compiler and the validation engine, the installation steps are as follows:
1. Create a virtualenv
1. activate the virtualenv
1. install the package (compiler through PyPI, validator through source code)
1. deactivate

The steps to use them are:

1. activate the respective virtualenv
1. issue the command line instruction to execute the compiler or the validator
1. deactivate the respective virtualenv

Reply:
My question is really: is a deactivate command needed right before a shell exits?

Reply:
Nope, you're right. I see your point. It is not needed. ;)